app/Rules/StringRule.php

```diff
@@ -26,25 +26,22 @@ public function __construct(bool $isNullable, int $limit = 0)
 	 */
 	public function passes($attribute, $value): bool
 	{
-		return
-			(
-				$value === null &&
-				$this->isNullable
-			) || (
-				is_string($value) &&
-				strlen($value) != 0 &&
-				($this->limit === 0 || strlen($value) <= $this->limit)
-			);
+		return ($value === null &&
+			$this->isNullable
+		) || (is_string($value) &&
+			strlen($value) != 0 &&
+			($this->limit === 0 || strlen($value) <= $this->limit)
+		);
 	}
 
 	/**
 	 * {@inheritDoc}
 	 */
-	public function message()
+	public function message(): string
 	{
 		return ':attribute must be' .
 			($this->isNullable ? ' either null or' : '') .
 			' a non-empty string' .
-			($this->limit ? ' with at most ' . $this->limit . ' characters' : '');
+			($this->limit != 0 ? ' with at most ' . $this->limit . ' characters' : '');
```

Comment:
```suggestion
			($this->limit !== 0 ? ' with at most ' . $this->limit . ' characters' : '');
```

Is there a test we can enable to warn about `==` / `!=`? :smiley:

Reply:
This is not really necessary if you have equality of types in both sides, further more this would include non desirable behaviors when comparing objects.

Reply:
What kind of undesirable behavior are you speaking about? I would also prefer strict comparison. See all my previous comments.